packages/api/src/utils/pin.js

```diff
@@ -108,3 +125,22 @@ export async function waitToGetOkPins (cid, cluster, waitTime = MAX_PIN_STATUS_C
   }
   return []
 }
+
+/**
+ * Used to async wait for pins for the provided cid to have an
+ * OK_STATUS and update them in the db.
+ *
+ * @param {string} cid
+ * @param {import('@nftstorage/ipfs-cluster').Cluster} cluster
+ * @param {import('@web3-storage/db').DBClient} db
+ * @param {number} waitTime
+ * @param {number} checkInterval
+ * @return {Promise.<import('@web3-storage/db/db-client-types').PinUpsertInput[]>}
+ */
+export async function waitAndUpdateOkPins (cid, cluster, db, waitTime = MAX_PIN_STATUS_CHECK_TIME, checkInterval = PIN_STATUS_CHECK_INTERVAL) {
+  const okPins = await waitOkPins(cid, cluster, waitTime, checkInterval)
+  for (const pin of okPins) {
+    await db.upsertPin(cid, pin)
```

Comment:
I know this was already like this before. But now that I see it, can we create an issue to have `upsertPins` receiving an array instead? What do you think? We would be able to upsert all of them in one request instead of multiple